src/js/view/popup/popupWrite.js

```diff
@@ -0,0 +1,372 @@
+        dropdownBtnIcon = document.getElementById(dropdownBtnClassName + 'Icon');
+        dropdownBtnTitle = document.getElementById(dropdownBtnClassName + 'Title');
+
+        if (dropdownBtnClassName === 'dropdownCalendarBtn') {
+            dropdownBtnIcon.style.backgroundColor = listItem.childNodes[0].style.backgroundColor;
+            this.currentCalendar = this.calendars[domutil.getData(listItem, 'index')];
+        }
+
+        dropdownBtnTitle.innerText = listItem.innerText;
+
+        this.focusedDropdown.style.display = 'none';
+        this.focusedDropdown = null;
+    }
+};
+
+PopupWrite.prototype._toggleAlldayCheck = function(target) {
+    var id = 'alldayCheck';
+
+    if ((target.id === id) || domutil.closest(target, '#' + id)) {
+        document.getElementById('popup-allday-checkbox').checked = !document.getElementById('popup-allday-checkbox').checked;
+    }
+};
+
+PopupWrite.prototype._toggleLockButton = function(target) {
+    var id = 'lockBtn';
+    var lockBtnIcon;
+
+    if ((target.id !== id) && !domutil.closest(target, '#' + id)) {
+        return;
+    }
+
+    lockBtnIcon = document.getElementById('lockBtnIcon');
+
+    if (domutil.hasClass(lockBtnIcon, 'ic-unlock')) {
+        lockBtnIcon.className = 'icon ic-lock';
+    } else {
+        lockBtnIcon.className = 'icon ic-unlock';
+    }
+};
+
+PopupWrite.prototype._onSaveSchedule = function(target) {
+    var subject, location, state, isAllday;
+
+    if (target.id !== 'saveBtn') {
+        return;
+    }
+
+    subject = document.getElementById('subject-input');
+
+    if (!subject.value) {
+        subject.focus();
+
+        return;
+    }
+
+    location = document.getElementById('location-input');
+    state = document.getElementById('dropdownStateBtnTitle');
+    isAllday = document.getElementById('popup-allday-checkbox');
+
+    this.fire('saveSchedule', {
+        calendar: this.currentCalendar,
+        subject: subject.value,
+        location: location.value,
+        startDate: this.rangePicker.getStartDate(),
+        endDate: this.rangePicker.getEndDate(),
+        state: state.innerText,
+        isAllday: isAllday.checked
+    });
+
+    this.hide();
+};
+
+/**
+ * @override
+ * @param {object} viewModel - view model from factory/monthView
+ */
+PopupWrite.prototype.render = function(viewModel) {
+    var calendars = this.calendars;
+    var layer = this.layer;
+    var self = this;
+    var pos;
+
+    if (calendars.length) {
+        viewModel.calendars = calendars;
+        viewModel.currentCalendar = this.currentCalendar = calendars[0];
+    }
+
+    layer.setContent(this.tmpl(viewModel));
+    this._createDatepicker(viewModel.start, viewModel.end);
+    layer.show();
+
+    pos = this._getRenderPosition(viewModel.guide.guideElements || [viewModel.guide.guideElement]);
+    layer.setPosition(pos.x, pos.y);
+    this._setArrowDirection(pos.arrow);
+
+    util.debounce(function() {
```

Comment:
Remove frequently attaching event listener in render function. Move it to constructor instead.